helm/cluster-azure/templates/_machine_pools.tpl

```diff
@@ -0,0 +1,81 @@
+{{- define "machine-pools" }}
+{{ range .Values.machinePools }}
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachinePool
+metadata:
+  annotations:
+    machine-pool.giantswarm.io/name: {{ include "resource.default.name" $ }}-{{ .name }}
+  labels:
+    giantswarm.io/machine-pool: {{ include "resource.default.name" $ }}-{{ .name }}
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ include "resource.default.name" $ }}-{{ .name }}
+  namespace: {{ $.Release.Namespace }}
+spec:
+  clusterName: {{ include "resource.default.name" $ }}
+  replicas: {{ .minSize }}
+  template:
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfig
+          name: {{ include "resource.default.name" $ }}-{{ .name }}
+      clusterName: {{ include "resource.default.name" $ }}
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: AzureMachinePool
+        name: {{ include "resource.default.name" $ }}-{{ .name }}
+      version: {{ $.Values.kubernetesVersion }}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachinePool
+metadata:
+  labels:
+    giantswarm.io/machine-pool: {{ include "resource.default.name" $ }}-{{ .name }}
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ include "resource.default.name" $ }}-{{ .name }}
+  namespace: {{ $.Release.Namespace }}
+spec:
+  location: {{ .availabilityZones }}
```

Comment:
is this correct / confusing? 

as far as i understand

`location` = `region containing all the availability Zones` 
`failureDomains` = `availability Zones over which to spread the nodes` - https://capz.sigs.k8s.io/topics/failure-domains.html#using-virtual-machine-scale-sets

Reply:
good catch - i've renamed to `location`